torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -1170,14 +1170,24 @@ std::unique_ptr<Fusion> SegmentedFusion::makeFusion(SegmentedGroup* sg) {
     fusion_segment->removeOutput(out);
   }
 
+  std::vector<TensorView*> view_tvs;
   for (auto inp : getAllInputs(sg)) {
-    fusion_segment->addInput(complete_to_segment_map.clone(inp));
+    auto clone_tv = complete_to_segment_map.clone(inp);
+    fusion_segment->addInput(clone_tv);
+    if (inp->isDefinition(ExprType::ViewOp)) {
+      TORCH_INTERNAL_ASSERT(clone_tv != nullptr && clone_tv->isA<TensorView>());
+      view_tvs.push_back(clone_tv->as<TensorView>());
+    }
   }
 
   for (auto out : getAllOutputs(sg)) {
     fusion_segment->addOutput(complete_to_segment_map.clone(out));
   }
 
+  for (auto tv : view_tvs) {
+    tv->convertRfactorToRootDomain();
```

Comment:
Yes, I was seeing an illegal memory access because the fusion input predicate was for the root domain instead of the rfactor domain.

It seems there is an implicit assumption that the fusion input does not have a rfactor domain.